### ISSUE TITLE
Screen Reader does not read the result when using ajax request

### ISSUE THREAD

**Issue by Anubarak** [select2/select2#6026]
When Using your [Ajax](https://select2.org/data-sources/ajax) example in 
* Chrome Version 89.0.4389.128 (Official Build) (64-bit) 
* NVDA Screen Reader 2020.4 

The screen reader does not read the results of the entries, instead it just re-rereads the input.
Steps to reproduce
* go to [Ajax](https://select2.org/data-sources/ajax)
* open a screen reader
* navigate to the select and type something
* try to navigate to the result -> hear the screen reader will always repeat what you types before but not the result

Reproduction Link:
[click](https://jsbin.com/kahawusiyi/edit?html,css,js,output)

```JS
$(document).ready(function() {
    $('#example1').select2({
      ajax: {
        url: 'https://thaff.2ndred.dev/',
        dataType: 'json',
        data: function (params) {
          return {
            action: 'thaff/zip-field/search-locations',
            states: ['123'],
            location: params.term
          };
        },
        processResults: function (data) {
          return {
            results: data.map(function (item) {
              return {
                id: item.zip + '*' + item.location,
                text: item.zip + ' ' + item.location
              };
            })
          };
        }
      },
      templateResult: function (item, container) {
        return item.text;
      },
      templateSelection: function (item, container) {
        return item.text;
      }
    });
});
```

**Comment by Anubarak** [comment]
Not fixed yet.. Please let it open bot

**Comment by kevin-brown** [comment]
Not sure how I forgot to tag this, my bad. Very aware of this issue, it was previously reported in https://github.com/select2/select2/issues/3735#issuecomment-780416916 because of an issue with the data not being properly normalized (which makes it accessible) in certain cases.

**Comment by kevin-brown** [comment]
Actually I'm going to close this as a duplicate of #3735 since it's the exact same root cause and problem reported.
